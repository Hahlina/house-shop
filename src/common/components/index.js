export * from './PropertyCard/PropertyCard';
export * from './PropertyCard/PropertyCardLoader';
export * from './SwiperSlide/SwiperSlide';
export * from './SwiperContainer/SwiperContainer';
export * from './NavBreadcrumbs/NavBreadcrumbs';
