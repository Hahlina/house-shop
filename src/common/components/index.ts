export * from './PropertyCard/PropertyCard';
export * from './PropertyCard/PropertyCardLoader';
export * from './SwiperSlide/SwiperSlide';
export * from './SwiperContainer/SwiperContainer';
export * from './NavBreadcrumbs/NavBreadcrumbs';
export * from './Container/Container';
export * from './Image/Image';
export * from './ErrorPage/ErrorPage';
export * from './SomethingWentWrong/SomethingWentWrong';
export * from './NothingFound/NothingFound';
