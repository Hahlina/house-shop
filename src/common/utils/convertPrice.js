export const convertPrice = (price) =>
    new Intl.NumberFormat('en-US', {
        style: 'currency',
        currency: 'USD'
    }).format(price || 0);
