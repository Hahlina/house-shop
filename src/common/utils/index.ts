export * from './convertPrice';
export * from './capitalizeFirstLetter';
