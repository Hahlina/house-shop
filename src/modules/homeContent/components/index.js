export * from "./PreviewBanner/PreviewBanner";
export * from "./PropertiesList/PropertiesList";
