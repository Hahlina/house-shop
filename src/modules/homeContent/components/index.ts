export * from './PreviewBanner/PreviewBanner';
export * from './PropertiesList/PropertiesList';
export * from './PropertiesPagination/PropertiesPagination';
