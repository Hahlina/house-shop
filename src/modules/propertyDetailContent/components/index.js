export * from "./PropertyGallery/PropertyGallery";
export * from "./FullInformationCard/FullInformationCard";
