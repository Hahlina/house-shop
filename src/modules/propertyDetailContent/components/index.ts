export * from './PropertyGallery/PropertyGallery';
export * from './PropertyGallery/PropertyGalleryLoader';
export * from './FullInformationCard/FullInformationCard';
export * from './FullInformationCard/FullInformationCardLoader';
export * from './PropertyDetailContentLoader/PropertyDetailContentLoader';
export * from './PropertyDetailBreadCrumbs/PropertyDetailBreadCrumbs';
