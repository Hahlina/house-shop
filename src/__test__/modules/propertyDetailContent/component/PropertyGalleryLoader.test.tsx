import { PropertyGalleryLoader } from 'modules/propertyDetailContent/components';

import { render, screen } from 'common/utils/test';

describe('FullInformationCardLoader', () => {
    it('renders the correct number of thumb skeletons components', () => {
        render(<PropertyGalleryLoader />);
        const thumbsSkeleton = screen.getAllByTestId('skeleton');
        expect(thumbsSkeleton).toHaveLength(5);
    });

    it('renders the correct height of thumb skeletons components', () => {
        render(<PropertyGalleryLoader />);
        const thumbsSkeleton = screen.getAllByTestId('skeleton');
        thumbsSkeleton.forEach((thumb) => {
            expect(thumb).toHaveStyle('height: 100px');
        });
    });

    it('renders the correct height of skeleton component for big photo', () => {
        render(<PropertyGalleryLoader />);
        const skeletonComponents = screen.getByTestId('main-photo');
        expect(skeletonComponents).toHaveStyle('height: 600px');
    });

    it('should take a snapshot', () => {
        const { container } = render(<PropertyGalleryLoader />);
        expect(container).toMatchSnapshot();
    });
});
