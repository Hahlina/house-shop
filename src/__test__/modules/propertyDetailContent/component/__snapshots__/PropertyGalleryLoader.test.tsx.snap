// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullInformationCardLoader should take a snapshot 1`] = `
<div>
  <div
    class="sc-qZrbh gHMCpY"
    data-testid="property-gallery-loader"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-153n0d3-MuiSkeleton-root"
      data-testid="main-photo"
      style="height: 600px;"
    />
    <div
      class="sc-irLvIq gVQkQG"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-153n0d3-MuiSkeleton-root"
        data-testid="skeleton"
        style="width: 100px; height: 100px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-153n0d3-MuiSkeleton-root"
        data-testid="skeleton"
        style="width: 100px; height: 100px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-153n0d3-MuiSkeleton-root"
        data-testid="skeleton"
        style="width: 100px; height: 100px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-153n0d3-MuiSkeleton-root"
        data-testid="skeleton"
        style="width: 100px; height: 100px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-153n0d3-MuiSkeleton-root"
        data-testid="skeleton"
        style="width: 100px; height: 100px;"
      />
    </div>
  </div>
</div>
`;
